go-controller/pkg/kube/annotator_test.go

```diff
@@ -0,0 +1,605 @@
+			expectedErr:            nil,
+			onRetMockInterfaceArgs: &ovntest.TestifyMockHelper{"CoreV1", []string{}, []interface{}{mockCoreIface}},
+			onRetMockCoreArgs:      &ovntest.TestifyMockHelper{"Nodes", []string{"string"}, []interface{}{mockNodeIface}},
+			onRetMockNodeArgs:      &ovntest.TestifyMockHelper{"Patch", []string{"*context.emptyCtx", "string", "types.PatchType", "[]uint8", "v1.PatchOptions"}, []interface{}{nil, nil}},
+			act:                    &action{key: "key"},
+		},
+		{
+			desc:                   "Negative test code path",
+			expectedErr:            fmt.Errorf("mock Error: Failed to run SetAnnotationsOnNode"),
+			onRetMockInterfaceArgs: &ovntest.TestifyMockHelper{"CoreV1", []string{}, []interface{}{mockCoreIface}},
+			onRetMockCoreArgs:      &ovntest.TestifyMockHelper{"Nodes", []string{"string"}, []interface{}{mockNodeIface}},
+			onRetMockNodeArgs:      &ovntest.TestifyMockHelper{"Patch", []string{"*context.emptyCtx", "string", "types.PatchType", "[]uint8", "v1.PatchOptions"}, []interface{}{nil, fmt.Errorf("mock Error: Failed to run SetAnnotationsOnNode")}},
+		},
+	}
+	for i, tc := range tests {
+
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			if tc.act != nil {
+				nodeAnnotator.changes[""] = tc.act
+			}
+
+			if tc.onRetMockInterfaceArgs != nil {
+				mockiFaceCall := mockIface.On(tc.onRetMockInterfaceArgs.OnCallMethodName)
+				for _, arg := range tc.onRetMockInterfaceArgs.OnCallMethodArgType {
+					mockiFaceCall.Arguments = append(mockiFaceCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetMockInterfaceArgs.RetArgList {
+					mockiFaceCall.ReturnArguments = append(mockiFaceCall.ReturnArguments, ret)
+				}
+				mockiFaceCall.Once()
+			}
+
+			if tc.onRetMockCoreArgs != nil {
+				mockCoreCall := mockCoreIface.On(tc.onRetMockCoreArgs.OnCallMethodName)
+				for _, arg := range tc.onRetMockCoreArgs.OnCallMethodArgType {
+					mockCoreCall.Arguments = append(mockCoreCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, elem := range tc.onRetMockCoreArgs.RetArgList {
+					mockCoreCall.ReturnArguments = append(mockCoreCall.ReturnArguments, elem)
+				}
+				mockCoreCall.Once()
+			}
+
+			if tc.onRetMockNodeArgs != nil {
+				mockPodCall := mockNodeIface.On(tc.onRetMockNodeArgs.OnCallMethodName)
+				for _, arg := range tc.onRetMockNodeArgs.OnCallMethodArgType {
+					mockPodCall.Arguments = append(mockPodCall.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range tc.onRetMockNodeArgs.RetArgList {
+					mockPodCall.ReturnArguments = append(mockPodCall.ReturnArguments, ret)
+				}
+				mockPodCall.Once()
+			}
+
+			e := nodeAnnotator.Run()
+
+			if tc.expectedErr != nil {
+				assert.Error(t, e)
+			} else {
+				assert.Nil(t, e)
+			}
+
+			mockIface.AssertExpectations(t)
+			mockCoreIface.AssertExpectations(t)
+			mockNodeIface.AssertExpectations(t)
+
+		})
+	}
+}
+
+func TestNewPodAnnotator(t *testing.T) {
```

Comment:
If the function only returns an object does it need to be tested in isolation like this?

Reply:
Helps with adding coverage and any future code edits